### PR TITLE
I18n

### PR DESCRIPTION
**Pull request by stonedDiscord**
the german translation works 🎉 
can't say what's going on with the japanese translation, i think qt changed the way it handles translations, because .mo files are nowhere to be found in the documentation

### CODE REVIEW

File: include/networkmanager.h [hunk 1]
```diff
@@ -14,6 +14,8 @@
 #undef MS_FAILOVER_SUPPORTED
 #endif
 
+#undef MS_FAILOVER_SUPPORTED
```
Comment (oldmud0) [inline include/networkmanager.h]: I don't think this was intentional on your end, but if it was, you might as well delete the failover code if you are going to do this.

**Comment by oldmud0** [comment]
What process do you have in mind for receiving translation contributions, distributing them, and loading them to AO? We can embed it as an RCC file inside the executable, load it from the default `translations` folder, or load it from `base/translations`.

File: include/networkmanager.h (hunk 1 above)
Comment (stonedDiscord) [reply include/networkmanager.h]: wasn't me

**Comment by stonedDiscord** [comment]
Qt Linguist ist actually a nice tool to translate something, so i would kindly ask contributors to download and use that.

Embedding the files makes a lot of sense as they will only need updating when a new feature is released.

The detection should default to the system locale if not set to something different in the settings.

**Comment by stonedDiscord** [comment]
they are now in the resource file

**Comment by stonedDiscord** [comment]
the thing about making it changeable is that the translations are applied before giving control to the app class.
need to look at a few other solutions to find out how i can do this right.

**Comment by likeawindrammer** [comment]
I got some compile errors while trying this: `cannot convert ‘QString’ to ‘const char*’ in assignment` and `no matching function for call to ‘QString::arg(std::__cxx11::basic_string<char>&)’` I did this https://github.com/likeawindrammer/AO2-Client/commit/e276ef2e51880b543c5fbb64e4b974652ec205ad to fix, but don't know if any of that is the correct way.

**Comment by stonedDiscord** [comment]
did you try it on linux or windows

**Comment by stonedDiscord** [comment]
https://wiki.qt.io/Technical_FAQ#How_can_I_convert_a_QString_to_char.2A_and_vice_versa.3F

need to do this

**Comment by stonedDiscord** [comment]
other option would be to just not localize the discord strings
i think other games do it like that

File: src/packet_distribution.cpp [hunk 2]
```diff
@@ -638,7 +639,7 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
   {
     if (courtroom_constructed && f_contents.size() >= 1)
     {
-      call_notice("You have been kicked from the server.\nReason: " + f_contents.at(0));
+      call_notice(tr("You have been kicked from the server.\nReason: ") + f_contents.at(0));
```
Comment (oldmud0) [inline src/packet_distribution.cpp]: You should still use the `%1` notation for this

**Comment by stonedDiscord** [comment]
did i forget anything?

**Comment by likeawindrammer** [comment]
> did i forget anything?

I need feedback for my translation, but it will be easier after the merge because of gitlab ci.
That aside I notice the language selector wont show the current selected language after save, it will always display "default".

**Comment by likeawindrammer** [comment]
Perfect

**Comment by stonedDiscord** [comment]
fingers crossed

**Comment by oldmud0** [comment]
Merge looks fine to me.

**Pull request merged**